File: PythonClient/ai.py
# -*- coding: utf-8 -*-

# python imports
import random

# chillin imports
from chillin_client import RealtimeAI

# project imports
from ks.models import World, Pacman, Ghost, Constants, ECell, EDirection
from ks.commands import ChangePacmanDirection, ChangeGhostDirection, ECommandDirection


class AI(RealtimeAI):

    def __init__(self, world):
        super(AI, self).__init__(world)


    def initialize(self):
        print('initialize')


    def decide(self):
        print('decide')
        
        if self.my_side == 'Pacman':
            # self.send_command(ChangePacmanDirection(direction=ECommandDirection.Down))
            direction = random.choice([
                ECommandDirection.Up,
                ECommandDirection.Right,
                ECommandDirection.Down,
                ECommandDirection.Left
            ])
            self.send_command(ChangePacmanDirection(direction=direction))

        if self.my_side == 'Ghost':
            direction = random.choice([
                ECommandDirection.Up,
                ECommandDirection.Right,
                ECommandDirection.Down,
                ECommandDirection.Left
            ])
            self.send_command(ChangeGhostDirection(direction=direction, id=0))
            self.send_command(ChangeGhostDirection(direction=direction, id=1))
            self.send_command(ChangeGhostDirection(direction=direction, id=2))
            # self.send_command(ChangeGhostDirection(direction=ECommandDirection.Up, id=0))
            # self.send_command(ChangeGhostDirection(direction=ECommandDirection.Right, id=1))

File: CppClient/Game/ai.cpp
#include "ai.h"

#include <iostream>
#include <vector>

#include "effolkronium/random.hpp"

using namespace std;
using Random = effolkronium::random_static;

using namespace koala::chillin::client;
using namespace ks::models;
using namespace ks::commands;


AI::AI(World *world): RealtimeAI<World*>(world)
{
}

AI::~AI()
{
}

void AI::initialize()
{
    cout << "initialize" << endl;
}

void AI::decide()
{
    cout << "decide" << endl;

    if (this->mySide == "Pacman")
    {
        auto random_direction = Random::get({EDirection::Up, EDirection::Right, EDirection::Down, EDirection::Left});
        changePacmanDirection(random_direction);
    }
    else if (this->mySide == "Ghost")
    {
        for (Ghost &ghost : this->world->ghosts())
        {
            auto random_direction = Random::get({EDirection::Up, EDirection::Right, EDirection::Down, EDirection::Left});
            changeGhostDirection(ghost.id(), random_direction);
        }
    }
}


void AI::changePacmanDirection(EDirection direction)
{
    ChangePacmanDirection cmd;
    cmd.direction((ECommandDirection) direction);
    this->sendCommand(&cmd);
}

void AI::changeGhostDirection(int id, EDirection direction)
{
    ChangeGhostDirection cmd;
    cmd.id(id);
    cmd.direction((ECommandDirection) direction);
    this->sendCommand(&cmd);
}
